### ISSUE TITLE
Experiment with new dependency analysis

### ISSUE THREAD

**Issue by oliviadargel** [amosproj/amos2021ws01-geo-data-search#195]
## User story
1. As a NLP Component developer
2. I want to improve the dependency analysis
3. So that I get better results

## Acceptance criteria
* result shared with NLP subteam 
* it is discussed with NLP subteam whether this draft should be used or not

## Definition of done
* Code was reviewed by another person
* GitHub CI runs successfully
* Feature is merged into main branch

**Comment by Battlemech** [comment]
Progress visible in https://github.com/amosproj/amos2021ws01-geo-data-search/tree/rework-get_dependencies(), will be merged once #196 is done
